src/context/useTransactionDialogContext.tsx

```diff
@@ -0,0 +1,118 @@
+import { TransactionNotifier } from '@synthetixio/transaction-notifier';
+import { Dialog, Spinner } from 'components';
+import TxStateSvg from 'components/Icons/TxState';
+import { providers } from 'ethers';
+import {
+  createContext,
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState
+} from 'react';
+
+export type TxState = 'signing' | 'mining' | 'failed' | 'confirmed' | 'error';
+
+interface TransactionDialogContextType {
+  setTxHash: Dispatch<string | undefined>;
+  visible: boolean;
+  setContent: Dispatch<SetStateAction<ReactNode | undefined>>;
+  state: TxState;
+  setVisible: Dispatch<SetStateAction<boolean>>;
+}
+
+const TransactionDialogContext = createContext<unknown>(null);
+
+export const useTransactionModalContext = () => {
+  return useContext(TransactionDialogContext) as TransactionDialogContextType;
+};
+const TransactionDialogContextProvider: React.FC<{
+  provider: providers.Web3Provider;
+  children: React.ReactNode;
+}> = ({ children, provider }) => {
+  const [visible, setVisible] = useState(false);
+  const [content, setContent] = useState<ReactNode | undefined>(undefined);
+  const [state, setState] = useState<TxState>('signing');
+  const [txHash, setTxHash] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
+    if (txHash) {
+      setVisible(true);
+      setState('signing');
+      const emitter = new TransactionNotifier(provider).hash(txHash);
+      emitter.on('txConfirmed', () => setState('confirmed'));
+      emitter.on('txError', () => setState('error'));
+      emitter.on('txFailed', () => setState('failed'));
+      emitter.on('txSent', () => setState('mining'));
+    }
+  }, [txHash, provider]);
+
+  const renderText = () => {
+    switch (state) {
+      case 'confirmed':
+        return (
+          <>
+            <h4 className="ui-tg-title-h4">Transaction Confirmed</h4>
+            <span className="ui-tg-caption ui-text-center ui-text-gray-500">
+              You can now proceed with the dApp.
+            </span>
+          </>
```

Comment:
we are repeating the tags and classes for each state.
so i think a better approach is to return { title: string, description: string} in a object and use it there